supporting_scripts/generate_code_cov_table/generate_code_cov_table.py

```diff
@@ -99,41 +96,51 @@ def filter_files(file_names):
     return client_file_names, server_file_names
 
 
-def get_client_line_coverage(username, password, key, file_name):
-    report_url = get_code_cov_report_url(username, password, key)
-    file_report_url = report_url.replace("index", file_name)
+def get_client_line_coverage(username, password, build_id, file_name):
+    report_url = get_client_tests_cov_report_url(build_id)
+    file_report_url = f"{report_url}/{file_name}.html"
     response = requests.get(file_report_url, auth=(username, password))
+    logging.debug(f"GET {file_report_url} -> {response.status_code}")
 
-    soup = BeautifulSoup(response.content, "html.parser")
-    coverage_divs = soup.find_all("div", {"class": "fl pad1y space-right2"})
     line_coverage = None
-    if len(coverage_divs) >= 4:
-        line_coverage_strong = coverage_divs[3].find("span", {"class": "strong"})
-        line_coverage = line_coverage_strong.text.replace("%", "").strip()
+    if response.status_code == 200:
+        soup = BeautifulSoup(response.content, "html.parser")
+        coverage_divs = soup.find_all("div", {"class": "fl pad1y space-right2"})
+        if len(coverage_divs) >= 4:
+            line_coverage_strong = coverage_divs[3].find("span", {"class": "strong"})
+            line_coverage = line_coverage_strong.text.replace("%", "").strip()
+        logging.debug(f"Coverage for {file_name} -> line coverage: {line_coverage}")
+    elif response.status_code != 404:
```

Comment:
Currently, the code covers only the cases when the status code is **200** or is **not 404**. Status code 404 itself is not handled. Why did you decide not to handle this case?

Reply:
Good question, so if the request is successful then the user is authorized to see the report and the report for this file exists => line coverage should be there 🤞
For when the file is not found I only saw status code `404` so far => line coverage is None since the file was deleted/not found

For all other status codes I don't know how to properly handle them or recover from them, I think just exiting with an error makes more sense instead of having a faulty report where maybe one line coverage is None because a request failed (bad internet etc.). Do you have a better solution in mind?

Reply:
Thank you for the response. I see your point now. I think you can keep it as it is 👍